tests/60app-services/05lookup3pe.pl

```diff
@@ -33,20 +68,80 @@
             assert_json_object( $body );
             assert_ok( defined $body->{ymca}, 'HS knows "ymca" protocol' );
 
+            my $ymca = $body->{ymca};
+            # sort the instances list for consistency
+            $ymca->{instances} = [ sort_by { $_->{desc} } @{ $ymca->{instances} } ];
+
+            assert_deeply_eq( $body->{ymca},
+               {
+                  user_fields     => [qw( field1 field2 )],
+                  location_fields => [qw( field3 )],
+                  icon            => "mxc://1234/56/7",
+                  instances       => [
+                     { desc => "instance 1" },
+                     { desc => "instance 2" },
+                     { desc => "instance 3" },
+                  ],
+               },
+               'fields in 3PE lookup metadata'
+            );
+
             Future->done(1);
          }),
       );
    };
 
+test "HS can provide query metadata on a single protocol",
+   requires => [ local_user_fixture(), $main::APPSERV[0], $main::APPSERV[1] ],
+
+   proves => [qw( can_get_3pe_metadata )],
+
+   check => sub {
+      my ( $user, $appserv1, $appserv2 ) = @_;
+
+      # Awkwardly, this test relies on the caching within synapse of the
+      # results obtained in the previous test.
```

Comment:
Why does it need to depend on the previous caching?

Could you explain what would be needed to avoid that dependency?